Scripts/RemediationScripts/Remediate-AddDiagnosticSettingsForLogicApps.ps1

```diff
@@ -0,0 +1,507 @@
+
+    if (-not (Test-Path -Path $backupFolderPath)) {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+
+    # Backing up Logic App details.
+    $backupFile = "$($backupFolderPath)\LogicAppsWithoutRequiredDiagnosticSetting.csv"
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 3 of 4] Backing up Logic App details to $($backupFile)"
+
+    $logicAppsWithoutRequiredDiagnosticSetting | Export-CSV -Path $backupFile -NoTypeInformation
+
+    if (-not $DryRun) {
+      
+        Write-Host $([Constants]::DoubleDashLine)
+        Write-Host "[Step 4 of 4] Adding Diagnostic setting for Logic Apps..." 
+
+        # To hold results from the remediation.
+        $remediatedLogicApps = @()
+
+        # Includes Logic Apps that were skipped during remediation due to any errors.
+        $skippedLogicApps = @()
+
+        # Storage Account details
+        [String] $storageAccountResourceGroupName = [String]::Empty
+        [String] $storageAccountName = [String]::Empty
+        $storageAccount = $null
+
+        
+        Write-Host "Adding diagnostic settng requires one or more of Storage Account, Log Analytics Workspace or Event Hub to be configured for storing diagnostic setting." -ForegroundColor $([Constants]::MessageType.Warning)
+        Write-Host "This script supports only Storage Accounts as a destination for storing the diagnostic settings." -ForegroundColor $([Constants]::MessageType.Warning)
+        Write-Host "In each resource group having Logic App resources, new Storage Accounts will be created per Resource Group and Location combination to store the diagnostic logs for all Logic Apps in the Resource Group to store diagnostic settings related data." -ForegroundColor $([Constants]::MessageType.Warning)
+        Write-Host "There will be billing cost associated with adding diagnostic setting." -ForegroundColor $([Constants]::MessageType.Warning)
+        Write-Host "Do you still want to proceed?" -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+
+        $userInput = Read-Host -Prompt "(Y|N)"
+
+        if ($userInput -ne "Y") {
+            Write-Host "Exiting as Storage Account is not chosen for storing the diagnostic logs..." -ForegroundColor $([Constants]::MessageType.Update)
+            break
+        }
+
+        $logicAppsWithoutRequiredDiagnosticSetting  | ForEach-Object {
+            Write-Host "Checking Logic App Resource Id $($_.ResourceId)" -ForegroundColor $([Constants]::MessageType.Info)
+
+            try {
+                $logicAppInstance = $_
+
+                $resourceGroup = $_.ResourceGroupName
+                $location = $_.Location
+                $storageAccountResourceGroupName = $resourceGroup
+                $storageAccountName = "diagn"
```

Comment:
Better to  generate some unique hash of  RG name + location + azts to avoid conflict.

Reply:
Updated as suggested, thanks